pkg/trainer/replicas.go

```diff
@@ -436,10 +432,49 @@ func (s *TFReplicaSet) GetStatus() (tfv1alpha1.TFReplicaStatus, error) {
 	return status, nil
 }
 
-func (s *TFReplicaSet) jobName(index int32) string {
+// SyncPods will try to check current pods for this TFReplicaSet and try to make it as desired.
+func (s *TFReplicaSet) SyncPods() error {
+	for index := int32(0); index < *s.Spec.Replicas; index++ {
```

Comment:
I think deleting failed pods could make debugging difficult. It will be useful to be able to see failed pods by doing 
kubectl describe pods <some pod> to get information about why it failed.

Instead of relying on the Name to identify pods belonging to a specific replica. Can we use the labels?
i.e with Get use a label selector to select on replica_index and replica_type you can then sort by creationtimestamp to get the most recent pod (this is what the current implementation does so you could probably copy much of the current logic).

The name for the pod can no include a random UUID. So when a pod fails we can just create a new one.
This would simplify some of the logic below since we no longer need to delete the pod; we can just check if one is running and if not we can create a new pod. So the logic is the same whether we are creating the pod for the first time or recovering.